crates/gitbutler-edit-mode/src/lib.rs

```diff
@@ -267,7 +267,11 @@ pub(crate) fn save_and_return_to_workspace(
     let parents = commit.parents().collect::<Vec<_>>();
 
     // Recommit commit
-    let tree = repository.create_wd_tree()?;
+    let mut index = repository.index()?;
```

Comment:
Oh, I think we have to look at `create_wd_tree()` then, and if this is a problem, then other usages of `create_wd_tree()` might also be.

The reason is that I changed it to bypass the index and instead work off a `git status` essentially. If this doesn't work or something is missing, then it would be good to either make work or revert the changes to `create_wd_tree()`.

Sorry for the hassle, I was convinced the implementation was equivalent.

CC @krlvi

Reply:
oh interesting... i remember struggling to create a workdir tree too for the oplog use case. it would be great if we could do that, without index add all

Reply:
I agree. To resolve this particular issue, I think I'd want to understand what `create_wd_tree()` didn't do anymore so hopefully that can be fixed. Maybe it's related to conflicts, and add-all automatically resolves them?